test/Altinn.Platform.Events.Tests/Utils/PrincipalUtil.cs

```diff
@@ -1,126 +1,221 @@
+    /// <summary>
+    /// Generates a token for an organization based on organization name, organization number, and optional scope.
+    /// </summary>
+    /// <param name="org">The organization name.</param>
+    /// <param name="orgNumber">The organization number. Default is "991825827".</param>
+    /// <param name="scope">The optional scope.</param>
+    /// <returns>A JWT token for the organization.</returns>
+    public static string GetOrgToken(string org, string orgNumber = "991825827", string scope = null)
+    {
+        ClaimsPrincipal principal = GetClaimsPrincipal(org, orgNumber, scope);
 
-            string token = JwtTokenMock.GenerateToken(principal, new TimeSpan(1, 1, 1));
+        return JwtTokenMock.GenerateToken(principal, new TimeSpan(1, 1, 1));
+    }
 
-            return token;
-        }
+    /// <summary>
+    /// Generates a token for a user based on user ID, authentication level, and optional scope.
+    /// </summary>
+    /// <param name="userId">The user ID.</param>
+    /// <param name="authenticationLevel">The authentication level. Default is 2.</param>
+    /// <param name="scope">The optional scope.</param>
+    /// <returns>A JWT token for the user.</returns>
+    public static string GetToken(int userId, int authenticationLevel = 2, string scope = null)
+    {
+        ClaimsPrincipal principal = GetClaimsPrincipal(userId, authenticationLevel, scope);
 
-        public static string GetToken(int userId, int authenticationLevel = 2, string scope = null)
-        {
-            ClaimsPrincipal principal = GetClaimsPrincipal(userId, authenticationLevel, scope);
+        return JwtTokenMock.GenerateToken(principal, new TimeSpan(0, 1, 5));
+    }
 
-            string token = JwtTokenMock.GenerateToken(principal, new TimeSpan(0, 1, 5));
+    /// <summary>
+    /// Generates a token for a system-user.
+    /// </summary>
+    /// <param name="systemId">The system identifier.</param>
+    /// <param name="systemUserId">The system user identifier.</param>
+    /// <param name="orgClaimId">The org claim identifier.</param>
+    /// <param name="authenticationLevel">The authentication level.</param>
+    /// <returns>A JWT token for the system-user.</returns>
+    public static string GetTokenForSystemUser(string systemId, string systemUserId, string orgClaimId, int authenticationLevel = 3)
+    {
+        ClaimsPrincipal principal = GetSystemUserPrincipal(systemId, systemUserId, orgClaimId, authenticationLevel);
 
-            return token;
-        }
+        return JwtTokenMock.GenerateToken(principal, new TimeSpan(0, 1, 5));
+    }
+
+    /// <summary>
+    /// Creates a ClaimsPrincipal object representing a system user.
+    /// </summary>
+    /// <param name="systemId">The system identifier.</param>
+    /// <param name="systemUserId">The system user identifier.</param>
+    /// <param name="orgClaimId">The organization claim identifier.</param>
```

Comment:
The org claim id value is actually the organization number of the system owner. We could name the parameter something like that instead.